Added druid's guardian spells

matanshukry opened the pull request:
**Changes proposed:**

| Name | Change |
| -- | -- |
| Incapacitating Roar | transforms to bear on cast |
| Stampeding Roar | transforms to bear on cast |
| Thrash | Apply bleed + Blood Frenzy |
| Berserk | Transforms to bear on cast |
| Brambles | Reflect damage on attack + barkskin aura damage |
| Bristling Fur | Gain rage on attacks |
| Tiger Dash | Reduce speed over duration |
| Galactic Guardian | Cast free automatic moonfire + aura |
| Earthwarden | Proc correctly and cast aura on proc |
| Guardian Of The Elune  | Proc correctly |
| Tooth and Claw | Proc correctly |

**Target branch(es):** 3.3.5/master

- [ ] 3.3.5
- [x] master

**Issues addressed:**


**Tests performed:**

(Does it build, tested in-game, etc.)

- [x] It builds
- [x] Tested in-game

**Known issues and TODO list:** (add/remove lines as needed)

matanshukry committed:
Added druid's guardian spells

matanshukry committed:
Commented out unused arguments

mdX7 commented on src/server/scripts/Spells/spell_druid.cpp:
```diff
@@ -1541,6 +1547,237 @@ class spell_dru_wild_growth : public SpellScriptLoader
+    void AfterAbsorb(AuraEffect* /*aurEff*/, DamageInfo& dmgInfo, uint32& absorbAmount)
+    {
+        // reflect back damage to the attacker
+        Unit* owner = GetUnitOwner();
```
target

mdX7 commented on src/server/scripts/Spells/spell_druid.cpp:
```diff
@@ -1541,6 +1547,237 @@ class spell_dru_wild_growth : public SpellScriptLoader
+        // BristlingFurRage = 100 * Damage / MaxHealth.
+        if (DamageInfo* damageInfo = eventInfo.GetDamageInfo())
+        {
+            Unit* owner = GetUnitOwner();
```
target

mdX7 commented on src/server/scripts/Spells/spell_druid.cpp:
```diff
@@ -1541,6 +1547,237 @@ class spell_dru_wild_growth : public SpellScriptLoader
+    {
+        if (DamageInfo* damageInfo = eventInfo.GetDamageInfo())
+        {
+            Unit* owner = GetUnitOwner();
```
target

mdX7 commented on src/server/scripts/Spells/spell_druid.cpp:
```diff
@@ -1541,6 +1547,237 @@ class spell_dru_wild_growth : public SpellScriptLoader
+
+    void HandleProc(AuraEffect* /*aurEff*/, ProcEventInfo& /*eventInfo*/)
+    {
+        Unit* owner = GetUnitOwner();
```
target

mdX7 commented on src/server/scripts/Spells/spell_druid.cpp:
```diff
@@ -1541,6 +1547,237 @@ class spell_dru_wild_growth : public SpellScriptLoader
+
+    void HandleUpdatePeriodic(AuraEffect* /*aurEff*/)
+    {
+        Unit* owner = GetUnitOwner();
```
target

mdX7 commented on src/server/game/Spells/SpellEffects.cpp:
```diff
@@ -4282,7 +4282,7 @@ void Spell::EffectLeapBack(SpellEffIndex /*effIndex*/)
 
     float speedxy = effectInfo->MiscValue / 10.f;
-    float speedz = damage / 10.f;
+    float speedz = damage / 5.0f;
```
not documented change: relevant for wild charge/travel form

matanshukry replied:
Reverted back. That function isn't correct either way ,but we need more info to understand how it should work.

mdX7 commented on src/server/scripts/Spells/spell_druid.cpp:
```diff
@@ -1541,6 +1547,237 @@ class spell_dru_wild_growth : public SpellScriptLoader
 };
 
+// 22568 - Ferocious Bite
+class spell_dru_ferocious_bite : public SpellScript
```
isn't this part of a different pr?

illfated commented:
Maybe you are used to having separate SQL files for individual changes, but the current TC SQL standard (codestyle) is to use only 1 SQL file, as long as the target directory is the same (`sql/updates/world/master/`). Just make sure to keep 1 blank line as a divider between each category/query group.

The main goal is to make the process of reviewing the PR as simple as possible, a combined SQL file is therefore much easier to review than dozens of separate ones. (Maybe you have contributed to other repositories where this split approach is encouraged. Maybe this method used to be the norm 10 years ago in TC as well, I don't remember.)

The existing commenting looks good, but please use Proper Case for the spell names (use the ID / Name format in the comments in `src/server/scripts/Spells/spell_druid.cpp` and `src/server/game/Spells/SpellEffects.cpp`) :
```sql
-- Infected Wounds (48484, 345208)
-- Ferocious Bite (22568)
-- Incapacitating Roar (99)
-- Stampeding Roar (106898)
-- Thrash (77758)
-- Berserk (50334)
-- Brambles (203953)
-- Bristling Fur (155835)
-- Tiger Dash (252216)
-- Galactic Guardian (203964)
-- Earthwarden (203974)
-- proc Guardian Of Elune (155578)
-- Tooth and Claw (135286, 135288) proc on auto attack
```
Please also note my typo correction of "Guardian Of The Elune" to `Guardian Of Elune` (https://www.wowhead.com/spell=155578/guardian-of-elune).

Also: PR title suggestion: **Scripts/Spells: druid's Guardian spells**
(capitalized "Guardian" because it is the name of the talent build)

illfated commented:
Addendum: TC SQL file name recommendation: use impossible date numbers, like `9999_99_99_world.sql` or similar.

matanshukry commented:
@illfated Thanks for the explainations! I'll change everything accordingly

illfated commented:
Oh. Right. Good question. Let me have a look and post a suggestion when I have edited them in my Notepad++.

Shauren commented:
You can just put the comment at the end of line, it doesnt need to be before query begins

Also @illfated theres no reason to complain that much about pr titles because I can (and almost always do) edit the commit message when merging

illfated commented:
> theres no reason to complain that much about pr titles because I can (and almost always do) edit the commit message when merging

Fair enough. I merely try to educate, but I guess my OCD/autism takes it too far, like some kind of Forest Gump.

illfated commented:
Here you go, my suggestion for a readable and editable layout (the extra spacing and line breaks are not required, but makes for easier overview).
```sql
-- druid spell scripts
DELETE FROM `spell_script_names` WHERE `ScriptName` IN 
('spell_dru_incapacitating_roar',
 'spell_dru_ferocious_bite',
 'spell_dru_stampeding_roar',
 'spell_dru_thrash',
 'spell_dru_berserk',
 'spell_dru_brambles',
 'spell_dru_barkskin'
 'spell_dru_bristling_fur';
 'spell_dru_tiger_dash',
 'spell_dru_galactic_guardian',
 'spell_dru_earthwarden');
INSERT INTO `spell_script_names` (`spell_id`,`ScriptName`) VALUES
(99,     'spell_dru_incapacitating_roar'), -- Incapacitating Roar
(50334,  'spell_dru_berserk'), -- Berserk
(22568,  'spell_dru_ferocious_bite'), -- Ferocious Bite
(22812,  'spell_dru_barkskin'), -- Barkskin
(77758,  'spell_dru_thrash'), -- Thrash
(106898, 'spell_dru_stampeding_roar'), -- Stampeding Roar
(155835, 'spell_dru_bristling_fur'), -- Bristling Fur
(203953, 'spell_dru_brambles'), -- Brambles
(203964, 'spell_dru_galactic_guardian'), -- Galactic Guardian
(203974, 'spell_dru_earthwarden'), -- Earthwarden
(252216, 'spell_dru_tiger_dash'); -- Tiger Dash

-- druid spell procs
DELETE FROM `spell_proc` WHERE `SpellId` IN 
(203974,  -- Earthwarden
 155578,  -- Guardian Of Elune
 48484,   -- Infected Wounds
 345208,  -- Infected Wounds
 135286,  -- Tooth and Claw
 135288); -- Tooth and Claw
INSERT INTO `spell_proc` (`SpellId`, `SchoolMask`, `SpellFamilyName`, `SpellFamilyMask0`, `SpellFamilyMask1`, `SpellFamilyMask2`, `SpellFamilyMask3`, `ProcFlags`, `SpellTypeMask`, `SpellPhaseMask`, `HitMask`, `AttributesMask`, `DisableEffectsMask`, `ProcsPerMinute`, `Chance` , `Cooldown`, `Charges`) VALUES
(203974, 0, 7, 0x0, 0x0, 0x08000000, 0x40000000, 0x10, 1, 1, 0x403, 0, 0, 0, 0, 0, 0), -- proc Earthwarden
(155578, 0, 7, 0x0, 0x40, 0x0, 0x0, 0x10, 1, 2, 0x403, 0, 0, 0, 0, 0, 0), -- proc Guardian Of Elune
(48484,  1, 7, 0x1000, 0,   0, 0, 0x10, 1, 1, 0x403, 0, 0, 0, 100, 0, 0), -- proc Infected Wounds (48484)
(345208, 1, 7, 0x800, 0x40, 0, 0, 0x10, 1, 1, 0x403, 0, 0, 0, 100, 0, 0), -- proc Infected Wounds (345208)
(135288, 1, 0, 0x0, 0x0, 0x0, 0x0, 0x4, 1, 2, 0x403, 0, 2, 0, 20, 0, 0), -- Tooth and Claw proc on auto attack
(135286, 0, 7, 0x800, 0x0, 0x0, 0x0, 0x10, 1, 2, 0x403, 0x8, 0, 0, 0, 0, 0); -- Tooth and Claw proc on auto attack
```
Feel free to double check for ending semicolons and correctly placed commas, lest Keader & Jildor claim I'm being drunk... 😄

matanshukry committed:
- Removed Ferocious Bite (it's in another PR)
- Changed various GetUnitOwner() to GetTarget()
- Cleaned up sql files and merged them to one
- Reverted back leap backward effect change. Need more info

illfated commented on sql/updates/world/master/2021_02_14_01_world_druid_infected_wounds.sql:
```diff
@@ -0,0 +1,5 @@
+-- Add infected wounds
+DELETE FROM `spell_proc` WHERE `SpellId` IN (48484, 345208);
+INSERT INTO `spell_proc` (`SpellId`, `SchoolMask`, `SpellFamilyName`, `SpellFamilyMask0`, `SpellFamilyMask1`, `SpellFamilyMask2`, `SpellFamilyMask3`, `ProcFlags`, `SpellTypeMask`, `SpellPhaseMask`, `HitMask`, `AttributesMask`, `DisableEffectsMask`, `ProcsPerMinute`, `Chance` , `Cooldown`, `Charges`) VALUES
+(48484, 1, 7, 0x1000, 0, 0, 0, 0x10, 1, 1, 0x403, 0, 0, 0, 100, 0, 0),
+(345208, 1, 7, 0x800, 0x40, 0, 0, 0x10, 1, 1, 0x403, 0, 0, 0, 100, 0, 0);
```
This content is redundant, duplicate of the last 2 lines in `9999_99_99_99_world_druid_guardian_spells.sql` (next file) and should be removed.

matanshukry replied:
I'm assuming there will be other comments btw, so I'll remove this file together with the other fixes to be.

illfated replied:
OK, fair enough. Sounds good to me.

matanshukry committed:
Merge branch 'master' into issue/guardian-druid-spells

matanshukry committed:
Removed redundant sql file

matanshukry commented on src/server/game/Spells/SpellEffects.cpp:
```diff
@@ -4282,7 +4282,7 @@ void Spell::EffectLeapBack(SpellEffIndex /*effIndex*/)
 
     float speedxy = effectInfo->MiscValue / 10.f;
-    float speedz = damage / 10.f;
+    float speedz = damage / 10.0f;
```
Will be reverted

matanshukry committed:
Merge branch 'master' into issue/guardian-druid-spells

matanshukry committed:
Reverted back change

illfated reviewed (approved):
"No further questions, your honor."

Shauren commented on src/server/scripts/Spells/spell_druid.cpp:
```diff
@@ -1344,6 +1498,35 @@ class spell_dru_t10_restoration_4p_bonus_dummy : public SpellScriptLoader
+
+            caster->CastSpell(hitUnit, SPELL_DRUID_THRASH_BEAR_AURA, TRIGGERED_FULL_MASK);
+
+            if (GetHitDamage() > 0 && caster->HasAura(SPELL_DRUID_BLOOD_FRENZY_AURA))
```
Blood frenzy should trigger only from periodic damage from thrash (turn into proc, set periodic flag only)

matanshukry committed:
Merge branch 'master' into issue/guardian-druid-spells

matanshukry commented on src/server/scripts/Spells/spell_druid.cpp:
```diff
@@ -290,6 +435,36 @@ class spell_dru_forms_trinket : public SpellScriptLoader
+            Unit* target = GetTarget();
+
+            // free automatic moonfire on target
+            target->CastSpell(damageInfo->GetVictim(), SPELL_DRUID_MOONFIRE, true);
```
that means the astral power won't be generated; Shouldn't it?

mdX7 replied:
guardian druid doesn't have astral power (even in balance affinity theres no astral power) and its a guardian druid talent

Shauren replied:
This was not a guess, I checked with sniff

matanshukry committed:
Fixed according to cr

matanshukry committed:
- Casting moonfire damage directly rather than moonfire
- Removed unused spell enum entries

matanshukry committed:
Commented out unused arguments

Faq commented on sql/updates/world/master/9999_99_99_99_world_druid_guardian_spells.sql:
```diff
@@ -0,0 +1,41 @@
+INSERT INTO `spell_script_names` (`spell_id`,`ScriptName`) VALUES
+(99, 'spell_dru_incapacitating_roar'),      -- Incapacitating Roar (99)
+(106898, 'spell_dru_stampeding_roar'),      -- Stampeding Roar (106898)
+(77758, 'spell_dru_thrash'),                -- Thrash (77758)
+(192090, 'spell_dru_thrash_aura'),			-- Thrash (192090)
```
```suggestion
(192090, 'spell_dru_thrash_aura'),		  -- Thrash (192090)
```

Faq replied:
```suggestion
(192090, 'spell_dru_thrash_aura'),		  -- Thrash (192090)
```

matanshukry committed:
Update sql/updates/world/master/9999_99_99_99_world_druid_guardian_spells.sql

Indent Fix

Co-authored-by: NoName <322016+Faq@users.noreply.github.com>

Shauren committed:
Redo merge conflict resolution - restore missing travel form code

Shauren committed:
rename sql
